lib/puppet/util/windows/adsi.rb

```diff
@@ -328,7 +335,14 @@ def sid
     def commit
       begin
         native_group.SetInfo unless native_group.nil?
-      rescue Exception => e
+      rescue WIN32OLERuntimeError => e
+        if e.message =~ /8007089A/m
+          raise Puppet::Error.new(
+            "Puppet is not able to create/delete domain groups with the group resource.\nPlease see http://links.puppetlabs.com/windows_usersgroups for details.",
+            e
+          )
+        end
+
```

Comment:
This actually changes the original code so that now we only capture `WIN32OLERuntimeError` instead of all exceptions... is that what we want?